src/debug_tools/stats.rs

```diff
@@ -0,0 +1,376 @@
+            ),
+            empty_rat: RatioStat::new(self.empty, other.empty),
+            hash_rat: RatioStat::new(self.hash, other.hash),
+            branch_rat: RatioStat::new(self.branch, other.branch),
+            extension_rat: RatioStat::new(self.extension, other.extension),
+            leaf_rat: RatioStat::new(self.leaf, other.leaf),
+        }
+    }
+}
+
+/// Information on the comparison between two tries.
+#[derive(Debug)]
+pub struct TrieComparison {
+    pub node_comp: NodeComparison,
+    pub depth_comp: DepthComparison,
+}
+
+impl Display for TrieComparison {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        writeln!(f, "Node comparison: {}", self.node_comp)?;
+        writeln!(f, "Depth comparison: {}", self.depth_comp)
+    }
+}
+
+// TODO: Consider computing these values lazily?
+#[derive(Debug)]
+pub struct NodeComparison {
+    pub tot_node_rat: RatioStat<usize>,
+    pub non_empty_rat: RatioStat<usize>,
+
+    pub empty_rat: RatioStat<usize>,
+    pub hash_rat: RatioStat<usize>,
+    pub branch_rat: RatioStat<usize>,
+    pub extension_rat: RatioStat<usize>,
+    pub leaf_rat: RatioStat<usize>,
+}
+
+impl Display for NodeComparison {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        writeln!(f, "Total nodes: {}", self.tot_node_rat)?;
+        writeln!(f, "Non-empty: {}", self.non_empty_rat)?;
+
+        writeln!(f, "Total empty: {}", self.empty_rat)?;
+        writeln!(f, "Total hash: {}", self.hash_rat)?;
+        writeln!(f, "Total branch: {}", self.branch_rat)?;
+        writeln!(f, "Total extension: {}", self.extension_rat)?;
+        writeln!(f, "Total leaf: {}", self.leaf_rat)
+    }
+}
+
+#[derive(Debug)]
+pub struct DepthComparison {
+    pub lowest_depth_rat: RatioStat<usize>,
+    pub avg_leaf_depth_rat: RatioStat<f32>,
+    pub avg_hash_depth_rat: RatioStat<f32>,
+}
+
+impl Display for DepthComparison {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        writeln!(f, "Lowest depth: {}", self.lowest_depth_rat)?;
+        writeln!(f, "Avg leaf depth: {}", self.avg_leaf_depth_rat)?;
+        writeln!(f, "Avg hash depth: {}", self.avg_hash_depth_rat)
+    }
+}
+
+/// Type to hold (and compare) a given variable from two different tries.s
+#[derive(Debug)]
+pub struct RatioStat<T> {
+    pub a: T,
+    pub b: T,
+}
+
+impl<T: Display + ToPrimitive> Display for RatioStat<T> {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        write!(
+            f,
+            "{:.3} / {:.3} ({:.3}%)",
+            self.a,
+            self.b,
+            self.get_a_over_b_perc()
+        )
+    }
+}
+
+impl<T: ToPrimitive> RatioStat<T> {
+    /// `new` doesn't have any logic, but this will reduce a lot of line lengths
+    /// since this is called so many times.
+    fn new(a: T, b: T) -> Self {
+        Self { a, b }
+    }
+
+    fn get_a_over_b_perc(&self) -> f32 {
+        (self.a.to_f32().unwrap() / self.b.to_f32().unwrap()) * 100.0
```

Comment:
I know it's only a debug tool but I'd rather expect() with a nice error message than just panicking

Reply:
Agreed, let me add this.